packages/api/src/payments/payrexxSubscriptionPaymentProvider.ts

```diff
@@ -0,0 +1,449 @@
+      const minPayment =
+        subscription.monthlyAmount * getMonths(subscription.paymentPeriodicity) - 100 // -1CHF to ensure that imported rounding differences are no issue
+      if (payedAmount < minPayment) {
+        logger('payrexxSubscriptionPaymentProvider').warn(
+          `Payrexx Subscription ${subscription.id} payment ${payedAmount} lower than min payment ${minPayment}`
+        )
+        return
+      }
+
+      // Delete unpaid
+      await deleteUnpaidInvoices(invoiceClient, subscriptionPeriodClient, subscription)
+
+      // Create invoice
+
+      const invoice = await invoiceClient.create({
+        data: {
+          mail: user.email,
+          dueAt: new Date(),
+          subscriptionID: subscription.id,
+          description: `Abo ${memberPlan.name}`,
+          paidAt: new Date(),
+          canceledAt: null,
+          sentReminderAt: new Date()
+        }
+      })
+
+      await invoiceItemClient.create({
+        data: {
+          invoiceId: invoice.id,
+          createdAt: new Date(),
+          modifiedAt: new Date(),
+          name: `Abo ${memberPlan.name}`,
+          quantity: 1,
+          amount: payedAmount
+        }
+      })
+      if (!invoice) throw Error('Cant create Invoice')
+
+      // Add subscription Period
+
+      const subscriptionPeriod = await subscriptionPeriodClient.create({
+        data: {
+          subscriptionId: subscription.id,
+          startsAt: longestPeriod.endsAt,
+          endsAt: newSubscriptionValidUntil,
+          paymentPeriodicity: subscription.paymentPeriodicity,
+          amount: payedAmount,
+          invoiceID: invoice.id
+        }
+      })
+      if (!subscriptionPeriod) throw Error('Cant create subscription period')
+
+      // Create Payment
+      const payment = await paymentClient.create({
+        data: {
+          paymentMethodID: subscription.paymentMethodID,
+          state: PaymentState.paid,
+          invoiceID: invoice.id
+        }
+      })
+      if (!payment) throw Error('Cant create Payment')
+
+      // Update subscription
+      await subscriptionClient.update({
+        where: {
+          id: subscription.id
+        },
+        data: {
+          paidUntil: newSubscriptionValidUntil
+        }
+      })
+      logger('payrexxSubscriptionPaymentProvider').info(
+        `Subscription ${subscription.id} for user ${user.email} successfully renewed.`
+      )
+    } else {
+      logger('payrexxSubscriptionPaymentProvider').info('External Auto renewal failed!')
+    }
+  }
+
+  async updateRemoteSubscription(subscriptionId: number, amount: string) {
+    const data = {
+      amount,
+      currency: 'CHF'
+    }
+    const signature = crypto
+      .createHmac('sha256', this.instanceAPISecret)
+      .update(qs.stringify(data))
+      .digest('base64')
+
+    const res = await fetch(
+      `https://api.payrexx.com/v1.0/Subscription/${subscriptionId}/?instance=${encodeURIComponent(
```

Comment:
We should add the base url (https://api.payrexx.com/v1.0/Subscription/) as a class property because it's reused on line 399. dry.